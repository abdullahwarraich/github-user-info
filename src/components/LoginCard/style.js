import styled from 'styled-components'

export const CardDiv = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  background: rgb(240, 240, 240);
  .ant-card {
    border-radius: 30px;
    box-shadow: 5px 5px 5px 5px #c2c2c2;
  }
`
