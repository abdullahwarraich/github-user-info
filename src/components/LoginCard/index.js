import React from 'react'
import { Card } from 'antd'
import { CardDiv } from './style'

const LoginCard = ({ children, title }) => (
  <CardDiv>
    <Card title={title}>{children}</Card>
  </CardDiv>
)

export default LoginCard
