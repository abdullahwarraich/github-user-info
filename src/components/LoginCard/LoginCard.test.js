import { render } from '@testing-library/react'
import LoginCard from './index'

describe('LoginCard', () => {
  let loginCardRender
  let title

  beforeEach(() => {
    title = 'Login Card Title'
  })

  afterEach(() => {
    loginCardRender.unmount()
  })

  test('should render LoginCard', async () => {
    loginCardRender = render(
      <LoginCard title={title}>
        <div>{'Login card child component'}</div>
        <div>{'Login card child component1'}</div>
      </LoginCard>
    )
    expect(loginCardRender).toMatchSnapshot()
  })
})
