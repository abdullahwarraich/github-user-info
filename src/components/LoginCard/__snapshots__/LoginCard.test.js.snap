// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginCard should render LoginCard 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdvvaa hvVISP"
      >
        <div
          class="ant-card ant-card-bordered"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                Login Card Title
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            <div>
              Login card child component
            </div>
            <div>
              Login card child component1
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdvvaa hvVISP"
    >
      <div
        class="ant-card ant-card-bordered"
      >
        <div
          class="ant-card-head"
        >
          <div
            class="ant-card-head-wrapper"
          >
            <div
              class="ant-card-head-title"
            >
              Login Card Title
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <div>
            Login card child component
          </div>
          <div>
            Login card child component1
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
